cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -122,6 +122,85 @@ function(ADD_THIRDPARTY_LIB LIB_NAME)
   endif()
 endfunction()
 
+# Based on MIT-licensed
+# https://gist.github.com/cristianadam/ef920342939a89fae3e8a85ca9459b49
```

Comment:
Could you add a comment saying what this does and why? This is a potential future head scratcher for whoever will debug this.

Reply:
Yes, I certainly will